python_modules/libraries/dagster-dbt/dagster_dbt/dbt_artifacts.py

```diff
@@ -0,0 +1,162 @@
+import logging
+import os
+import shutil
+from pathlib import Path
+from typing import List, Optional, Union, cast
+
+from dagster._annotations import public
+
+from .core.resources_v2 import DbtCliResource
+from .errors import DagsterDbtManifestNotPreparedError, DagsterDbtProjectNotFoundError
+
+
+class _NullObj:
+    def __getattr__(self, name):
+        return lambda *args, **kwargs: None
+
+
+_NULL_LOGGER = cast(logging.Logger, _NullObj())
+
+
+class DbtArtifacts:
+    def __init__(
+        self,
+        project_dir: Union[Path, str],
+        *,
+        target_folder: Union[Path, str] = "target",
+        prepare_command: List[str] = ["parse", "--quiet"],
+        package_data_dir: Optional[Union[Path, str]] = None,
```

Comment:
For some reason this path shuffling stuff was extremely hard to wrap my head around, and I feel like the weirdness here is potentially stemming from the naming here.

Is it correct that you'd only want to set a package_data_dir in the case that your project_dir is something that is in a different directory than your dagster code? My feeling is that this should be called `deployed_project_dir` or something of the sort, and the current-day `project_dir` parameter can be called `local_project_dir` (at least internally, the user-facing parameter can be the same)

Then, there could be a single property `current_project_dir` which returns either `self._local_project_dir` or `self._deployed_project_dir` depending on the context (i.e.  if `DAGSTER_IS_DEV_CLI` is not set and self._deployed_project_dir is non-None, return deployed dir, otherwise return current dir)

This is almost, but not quite, what self._base_dir is, but I feel like that property muddies the waters a bit and could be removed entirely, as it's only used in `get_cli_resource` (which could become trivial with that property) and setting the manifest path (which could also use the current_project_dir, as far as I can tell)